### ISSUE TITLE
WebSocket is creating an Exception / Exception flow out of normal code in test case with Blazor Server Side

### ISSUE THREAD

**Issue by seanjbsol** [Sicos1977/ChromeHtmlToPdf#36]
Hi there,

A very exciting project, and very useful considering wkhtmltopdf does not support HTML 5 and renders, in my opinion, a poorer quality document than Chrome does with print to PDF. I'm also frustrated that I cannot seem to get FontAwesome Icons to display consistently in the rendered output.

I tested your example solution in isolation and it worked fine (running on IIS Express on my dev host), but when I've tried to get it to work with a URL on a Blazor server app (the page to convert is https://www.fleet-serv.app/Reports/RCI/13 - this is a razor page, not one that uses SignalR), it throws an exception in the hub circuit (I've try--catched the block where the conversion is happening and no Exception is caught) When I've managed to get an exception, it's from the WebSocket connection OnError method, which states that a Message was interrupted, but I cannot find where in the code there is any interruption to this, and this only happens when I'm breaking the execution at every message send and receive. 

I've tried various options:
1) Changing the default parameters to run Chrome in Headful mode - does not work at all
2) Adding a responseType of Base64Encoded string (or whatever it was from the chrome headless dev-tools API) did not change the behaviour at all
3) It's after the Page.printToPDF Message is sent. It appears the Task is not set to a Task.CompletedTask as the Exception happens outside of the code flow I can see.
4) I've tried running the debugger in Chrome and in Edge, with Script Debugging enabled and disabled and get the same result.

I will look into this in more detail when time allows, I'll possibly need to bring in the source for WebSocketSharp into the debugger to get a better feel for where the exception is being thrown.

I've tried running in the debugger both as Kestrel and as IIS Express. I've also changed the target of the https request in the Web API sample, which works fine.

So it looks like it's a peculiarity of working with Blazor and SignalR at this point,.

I will need to refactor the entire project in future to work with Blazor WebAssembly, to enable offline use and use in areas where signal connection is sub-optimal, but that's a way away yet, and I had hoped that this very promising project would be my silver bullet.

Thanks for all your hard work on this.

Thanks
Sean

**Comment by seanjbsol** [comment]
**UPDATE**:

I've moved the code to generate the PDF from within the circuit / component lifecycle to a separate API (MVC Controller) endpoint and it works fine (apart from background colours!), so the issue seems to be a WebSocket conflict, or something specific to Blazor.

Now I've just got to get the rest of it rendering right so I can move all the reports to use an MVC endpoint rather than the ReportingService class I wrote!

I'll leave the issue open if you want to investigate further, but I've got a workaround for my reasons now.

Thanks
